examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py

```diff
@@ -110,36 +116,39 @@ def __process_work_order_get_result(self, wo_id, jrpc_id, response):
             - response is the response object to be returned
         """
 
+        wo_id = params["workOrderId"]
         # Work order is processed if it is in wo-response table
         value = self.kv_helper.get("wo-responses", wo_id)
         if value:
             response = json.loads(value)
-            if "error" in response:
-                # Mapping standard enclave error codes to JSON RPC error codes
-                if response["error"]["code"] == EnclaveError.ENCLAVE_ERR_VALUE:
-                    response["error"]["code"] = WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE
-                elif response["error"]["code="] == EnclaveError.ENCLAVE_ERR_UNKNOWN:
-                    response["error"]["code"] = WorkOrderStatus.UNKNOWN_ERROR
-                else:
-                    response["error"]["code"] = WorkOrderStatus.FAILED
-        else:
-            if(self.kv_helper.get("wo-timestamps", wo_id) is not None):
-                # work order is yet to be processed
-                response = utility.create_error_response(
-                    WorkOrderStatus.PENDING, jrpc_id,
-                    "Work order result is yet to be updated")
+            if 'result' in response:
+                return response['result']
+
+            # response without a result should have an error
+            err = response["error"]
+            err_code = err["code"]
+            if err_code == EnclaveError.ENCLAVE_ERR_VALUE:
+                err["code"] = WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE
+            elif err_code == EnclaveError.ENCLAVE_ERR_UNKNOWN:
+                err["code"] = WorkOrderStatus.UNKNOWN_ERROR
             else:
-                # work order not in 'wo-timestamps' table
-                response = utility.create_error_response(
-                    WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE,
-                    jrpc_id,
-                    "Work order Id not found in the database. \
-                     Hence invalid parameter")
+                err["code"] = WorkOrderStatus.FAILED
+
+            return err
 
-        return response
+        if(self.kv_helper.get("wo-timestamps", wo_id) is not None):
+            # work order is yet to be processed
+            raise JSONRPCDispatchException(
```

Comment:
Once work order accepts work request it should send response as work order processing is pending. It is good return response instead of throwing exception

Reply:
Once the relevant changes are done in examples/common/python/connectors/direct/tcs_listener/tcs_listener.py", tcs_listener.py can catch this exception

Reply:
Currently, exceptions of this kind would be handled gracefully by the dispatcher  at line 53 and transformed into correct error responses, which is consistent with the previous hand-coded version~ @Ram-srini

Reply:
@sammyne.  Instead of doing changes to work order handler, can JSON RPC dispatcher be used i at one level above in tcs_listener.py ?

Reply:
Of course. If you don't mind the big PR, I'll do~ @manojgop

Reply:
@sammyne My understanding is changes will be small and  similar to this PR. But as a first step instead of making modification in individual handlers , JSON RPC dispatcher will be implemented in tcs_listener.py ?

Reply:
done as ad2b2e32b6919c36a50cddb4e18d1818534fb737